wai-handler-hal/src/Network/Wai/Handler/Hal.hs

```diff
@@ -170,12 +173,21 @@ toWaiRequest vault port req = do
             NS.addrFamily = NS.AF_INET,
             NS.addrSocketType = NS.Stream
           }
-      sourceIp =
+      source =
         T.unpack
           . HalRequest.sourceIp
           . HalRequest.identity
           $ HalRequest.requestContext req
-  sourceAddr : _ <- NS.getAddrInfo (Just hints) (Just sourceIp) (Just $ show port)
+  -- Test invokes from the API Gateway console pass a "sourceIp" field
+  -- of "test-invoke-source-ip". If the getAddrInfo call fails, just
+  -- assume localhost.
+  sourceHost <-
+    tryJust
+      (Just @IOException)
+      (NS.getAddrInfo (Just hints) (Just source) (Just $ show port))
+      <&> \case
+        Right (s : _) -> NS.addrAddress s
+        _ -> NS.SockAddrInet port $ NS.tupleToHostAddress (127, 0, 0, 1)
```

Comment:
Feels risky as it replaces fail-fast behavior with a "pass silently" behavior. Its obvious that proper fix should be done at AWS's test tool side, and realistically this never happens so I'd at least issue a warning here.